exchanges/coinbene/coinbene.go

```diff
@@ -0,0 +1,342 @@
+		c.AvailablePairs = exch.AvailablePairs
+		c.EnabledPairs = exch.EnabledPairs
+		err := c.SetCurrencyPairFormat()
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = c.SetAssetTypes()
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = c.SetAutoPairDefaults()
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = c.SetAPIURL(exch)
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = c.SetClientProxyAddress(exch.ProxyAddress)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		err = c.Websocket.Setup(c.WsConnect,
+			c.Subscribe,
+			c.Unsubscribe,
+			exch.Name,
+			exch.Websocket,
+			exch.Verbose,
+			coinbeneWsURL,
+			exch.WebsocketURL,
+			exch.AuthenticatedWebsocketAPISupport)
+		if err != nil {
+			log.Fatal(err)
+		}
+		c.WebsocketConn = &wshandler.WebsocketConnection{
+			ExchangeName:         c.Name,
+			URL:                  c.Websocket.GetWebsocketURL(),
+			ProxyURL:             c.Websocket.GetProxyAddress(),
+			Verbose:              c.Verbose,
+			ResponseCheckTimeout: exch.WebsocketResponseCheckTimeout,
+			ResponseMaxLimit:     exch.WebsocketResponseMaxLimit,
+		}
+		c.Websocket.Orderbook.Setup(
+			exch.WebsocketOrderbookBufferLimit,
+			true,
+			true,
+			false,
+			false,
+			exch.Name)
+	}
+}
+
+// FetchTicker gets and stores ticker data for a currency pair
+func (c *Coinbene) FetchTicker(symbol string) (TickerResponse, error) {
+	var t TickerResponse
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	path := fmt.Sprintf("%s%s%s?%s", c.APIUrl, coinbeneAPIVersion, coinbeneFetchTicker, params.Encode())
+	return t, c.SendHTTPRequest(path, &t)
+}
+
+// FetchOrderbooks gets and stores orderbook data for given pair
+func (c *Coinbene) FetchOrderbooks(symbol string, size int64) (OrderbookResponse, error) {
+	var o OrderbookResponse
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	intSize := strconv.FormatInt(size, 10)
+	params.Set("depth", intSize)
+	path := fmt.Sprintf("%s%s%s?%s", c.APIUrl, coinbeneAPIVersion, coinbeneFetchOrderBook, params.Encode())
+	return o, c.SendHTTPRequest(path, &o)
+}
+
+// GetTrades gets recent trades from the exchange
+func (c *Coinbene) GetTrades(symbol string) (TradeResponse, error) {
+	var t TradeResponse
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	path := fmt.Sprintf("%s%s%s?%s", c.APIUrl, coinbeneAPIVersion, coinbeneGetTrades, params.Encode())
+	return t, c.SendHTTPRequest(path, &t)
+}
+
+// GetPairInfo gets info about a single pair
+func (c *Coinbene) GetPairInfo(symbol string) (PairResponse, error) {
+	var resp PairResponse
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	path := fmt.Sprintf("%s%s%s?%s", c.APIUrl, coinbeneAPIVersion, coinbenePairInfo, params.Encode())
+	return resp, c.SendHTTPRequest(path, &resp)
+}
+
+// GetAllPairs gets all pairs on the exchange
+func (c *Coinbene) GetAllPairs() (AllPairResponse, error) {
+	var a AllPairResponse
+	path := fmt.Sprintf("%s%s%s", c.APIUrl, coinbeneAPIVersion, coinbeneGetAllPairs)
```

Comment:
When you're just adding strings, please use +.

Coming from the horse's mouth:

"It is usually clearer and always faster to use the built-in string comparison operators ==, <, >, and so on."